editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/CamelDesignEditor.java

```diff
@@ -282,28 +284,34 @@ protected FlyoutPaletteComposite createPaletteComposite(Composite parent) {
 	@Override
 	public void selectionChanged(IWorkbenchPart part, ISelection selection) {
 		// If not the active editor, ignore selection changed.
-		if (getSite().getPage().getActiveEditor() == this.parent && this.parent.getActiveEditor() == this) {
-			updateActions(getSelectionActions());
+		IWorkbenchPartSite site = getSite();
+		if (site != null) {
+			IWorkbenchPage page = site.getPage();
+			if (page != null
+					&& page.getActiveEditor() == this.parent
```

Comment:
not really, parent is initialized in the constructor. If it is null, it will fails before reaching this point. There is no assignment outside of the constructor.